github/TeamDiscussion.py

```diff
@@ -0,0 +1,211 @@
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+from __future__ import absolute_import
+
+import github.GithubObject
+import github.NamedUser
+
+
+class TeamDiscussion(github.GithubObject.NonCompletableGithubObject):
+    """
+    This class represents TeamDiscussions. The reference can be found here https://developer.github.com/v3/teams/discussions/
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"number": self._number.value, "title": self._title.value})
+
+    @property
+    def author(self):
+        """
+        :type: :class:`github.NamedUser.NamedUser`
+        """
+        return self._author.value
+
+    @property
+    def body(self):
+        """
+        :type: string
+        """
+        return self._body.value
+
+    @property
+    def body_html(self):
+        """
+        :type: string
+        """
+        return self._body_html.value
+
+    @property
+    def body_version(self):
+        """
+        :type: string
+        """
+        return self._body_version.value
+
+    @property
+    def comments_count(self):
+        """
+        :type: integer
+        """
+        return self._comments_count.value
+
+    @property
+    def comments_url(self):
+        """
+        :type: string
+        """
+        return self._comments_url.value
+
+    @property
+    def created_at(self):
+        """
+        :type: datetime.datetime
+        """
+        return self._created_at.value
+
+    @property
+    def html_url(self):
+        """
+        :type: string
+        """
+        return self._html_url.value
+
+    @property
+    def last_edited_at(self):
+        """
+        :type: datetime.datetime
+        """
+        return self._last_edited_at.value
+
+    @property
+    def node_id(self):
+        """
+        :type: string
+        """
+        return self._node_id.value
+
+    @property
+    def number(self):
+        """
+        :type: integer
+        """
+        return self._number.value
+
+    @property
+    def pinned(self):
+        """
+        :type: bool
+        """
+        return self._pinned.value
+
+    @property
+    def private(self):
+        """
+        :type: bool
+        """
+        return self._private.value
+
+    @property
+    def team_url(self):
+        """
+        :type: string
+        """
+        return self._team_url.value
+
+    @property
+    def title(self):
+        """
+        :type: string
+        """
+        return self._title.value
+
+    @property
+    def updated_at(self):
+        """
+        :type: datetime.datetime
+        """
+        return self._updated_at.value
+
+    @property
+    def url(self):
+        """
+        :type: string
+        """
+        return self._url.value
```

Comment:
These are all missing self._completeIfNotSet(<attribute>)

Reply:
self._completIfNotSet(self._url) for that one, and the same pattern for the others, since GitHub ate my markup with angle brackets. Bad GitHub, no cookie.

Reply:
I don't think those calls are needed since `TeamDiscussion` is a child of `NonCompletableGithubObject`.

Reply:
Ah, I missed that. However, since TeamDiscussions have a URL, I think they are completable. Can you check?

Reply:
Good catch! I added the calls to `_completeIfNotSet` and changed the parent class.